sage_modabvar/abvar.py

```diff
@@ -2129,6 +2130,42 @@ def _ambient_hecke_matrix_on_modular_symbols(self, n):
         self.__ambient_hecke_matrix_on_modular_symbols[n] = T
         return T
 
+    def number_of_rational_points(self):
+        """
+        Return the number of rational points of this modular abelian variety.
+
+        It is not always possible to compute the order of the torsion
+        subgroup. The BSD conjecture is assumed to compute the algebraic rank.
+
+        OUTPUT: a positive integer or infinity
+
+        EXAMPLES::
+
+            sage: from sage_modabvar import J0, J1
+            sage: J0(23).number_of_rational_points()
+            11
+            sage: J0(29).number_of_rational_points()
+            7
+            sage: J0(37).number_of_rational_points()
+            +Infinity
+            sage: J1(17).number_of_rational_points()
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: torsion multiple only implemented for Gamma0
+        """
+        # Check easy dimension zero case
+        if self.dimension() == 0:
+            return ZZ(0)
```

Comment:
1. This is **wrong**.  A 0-dimensional abelian variety has one point on it, namely 0.  

2. There should be a doctest illustrating this case (correctly).